### PR TITLE
Better SVG path parsing

### PR DESCRIPTION
**Pull request by achretien**
Fix an error when parsing the following 'a' command $arguments:

5.38022,5.38022,0,0,1-2.4207.72246,4.50524,4.50524,0,0,1-3.12681-1.33942

Old parsing (error):
-------------------
First item:
- a[0][0] => 5.38022
- a[0][1] => 5.38022
- a[0][2] => 0
- a[0][3] => 0
- a[0][4] => 1
- a[0][5] => -2.4207.72246
- a[0][6] => 4.50524

Second item:
- a[0][7] => 4.50524
- a[0][8] => 0
- a[0][9] => 0
- a[0][10] => 1
- a[0][11] => -3.12681
- a[0][12] => -1.33942

New parsing (OK):
-----------------------
First item:
- a[0] => 5.38022
- a[1] => 5.38022
- a[2] => 0
- a[3] => 0
- a[4] => 1
- a[5] => -2.4207
- a[6] => .72246

Second item:
- a[7] => 4.50524
- a[8] => 4.50524
- a[9] => 0
- a[10] => 0
- a[11] => 1
- a[12] => -3.12681
- a[13] => -1.33942

Source:
--------

https://github.com/PhenX/php-svg-lib/blob/master/src/Svg/Tag/Path.php#L47

### CODE REVIEW

File: classes/svg.php [hunk 1]
```diff
@@ -1826,7 +1829,8 @@ function Arcto($x1, $y1, $x2, $y2, $rx, $ry, $angle, $largeArcFlag, $sweepFlag)
 // If rX = 0 or rY = 0 then this arc is treated as a straight line
 		// segment (a "lineto") joining the endpoints.
 		if ($rx == 0.0 || $ry == 0.0) {
-			//   return array(Lineto(x2, y2), $bounds);
+			return array('', $bounds); // mPD 5.0.040
+			return array(Lineto(x2, y2), $bounds);
```
Comment (finwe) [inline classes/svg.php]: Second return is never reached.

**Comment by finwe** [comment]
Thanks for your contribution. Please, base your PRs on `development` branch. Will look into it when based correctly.

File: classes/svg.php (hunk 1 above)
Comment (achretien) [inline classes/svg.php]: This line is a test I forgot to remove.

**Comment by achretien** [comment]
Before doing that, I would like to know the politic because there is no release yet based on the new (better) code structure, isn't it?

I'm using a lib based on your 6.x version so you can update the dev if you want with the code in my PR or use the SVG lib I linked in the comment.

**Comment by finwe** [comment]
master/6.x branches are frozen in favor of finalizing 7.x version. Given changes in code for development branch, it gives too big overhead to maintain both (with an exception for critical issues).

Does your PR provide substantial enhancement in functionality in particular use cases or is it - as it seems - only an aesthetic enhancement?

Either way, I'll be happy to review it when it is based on development branch. Feel free to reopen this if ready.

It would also be nice if you added some unit tests along with the change.

**Comment by achretien** [comment]
In fact it is a bug fix.

The current parsing on the given path is wrong in your version

5.38022,5.38022,0,0,1 `-2.4207.72246` ,4.50524,4.50524,0,0,1-3.12681-1.33942

- a[0][5] => `-2.4207.72246`
- a[0][6] => `4.50524`

against

- a[5] => `-2.4207`
- a[6] => `.72246`

**Comment by finwe** [comment]
I'll be even more happy to review this when based on development branch then! :)

By the way, is a decimal with multiple decimal separators even a thing in Svg? How does that work? Is there a spec for that?

**Comment by Mouradif** [comment]
I was just about to post an issue regarding SVG path parsing, I thinkg this PR would solve my case, I have an SVG logo on top of invoices generated with mpdf ([this logo](https://anywhere.livingparisian.com/assets/images/logo_logosvg.svg)) I get the following errors after calling

`$pdf->writeHTML($html)` : 

```
<br />
<b>Notice</b>:  Undefined offset: 5 in <b>/vendor/mpdf/mpdf/src/Image/Svg.php</b> on line <b>1734</b><br />
<br />
<b>Notice</b>:  Undefined offset: 5 in <b>/vendor/mpdf/mpdf/src/Image/Svg.php</b> on line <b>1734</b><br />
<br />
<b>Notice</b>:  Undefined offset: 6 in <b>/vendor/mpdf/mpdf/src/Image/Svg.php</b> on line <b>1887</b><br />
<br />
<b>Notice</b>:  Undefined offset: 6 in <b>/vendor/mpdf/mpdf/src/Image/Svg.php</b> on line <b>1894</b><br />
<br />
<b>Notice</b>:  Undefined offset: 6 in <b>/vendor/mpdf/mpdf/src/Image/Svg.php</b> on line <b>1887</b><br />
<br />
<b>Notice</b>:  Undefined offset: 6 in <b>/vendor/mpdf/mpdf/src/Image/Svg.php</b> on line <b>1894</b><br />
<br />
<b>Notice</b>:  Undefined offset: 13 in <b>/vendor/mpdf/mpdf/src/Image/Svg.php</b> on line <b>1887</b><br />
<br />
<b>Notice</b>:  Undefined offset: 13 in <b>/vendor/mpdf/mpdf/src/Image/Svg.php</b> on line <b>1894</b><br />
<br />
<b>Notice</b>:  Undefined offset: 1 in <b>/vendor/mpdf/mpdf/src/Image/Svg.php</b> on line <b>1577</b><br />
<br />
<b>Notice</b>:  Undefined offset: 1 in <b>/vendor/mpdf/mpdf/src/Image/Svg.php</b> on line <b>1577</b><br />
<br />
<b>Notice</b>:  Undefined offset: 6 in <b>/vendor/mpdf/mpdf/src/Image/Svg.php</b> on line <b>1887</b><br />
<br />
<b>Notice</b>:  Undefined offset: 6 in <b>/vendor/mpdf/mpdf/src/Image/Svg.php</b> on line <b>1894</b><br />
<br />
<b>Notice</b>:  Undefined offset: 6 in <b>/vendor/mpdf/mpdf/src/Image/Svg.php</b> on line <b>1887</b><br />
<br />
<b>Notice</b>:  Undefined offset: 6 in <b>/vendor/mpdf/mpdf/src/Image/Svg.php</b> on line <b>1894</b><br />
<br />
<b>Notice</b>:  Undefined offset: 21 in <b>/vendor/mpdf/mpdf/src/Image/Svg.php</b> on line <b>1732</b><br />
<br />
<b>Notice</b>:  Undefined offset: 22 in <b>/vendor/mpdf/mpdf/src/Image/Svg.php</b> on line <b>1733</b><br />
<br />
<b>Notice</b>:  Undefined offset: 23 in <b>/vendor/mpdf/mpdf/src/Image/Svg.php</b> on line <b>1734</b><br />
```

**Comment by finwe** [comment]
There is an open PR on current development branch https://github.com/mpdf/mpdf/pull/519

**Comment by finwe** [comment]
@Mouradif Can you test your SVG file with the code from mentioned PR?

**Pull request closed without merging**